vars/daosLatestVersion.groovy

```diff
@@ -0,0 +1,55 @@
+/* groovylint-disable DuplicateNumberLiteral, ParameterName, VariableName */
+// vars/daosLatestVersion.groovy
+
+  /**
+   * daosLatestVersion step method
+   *
+   * @param version search ceiling
+   * @param repository type
+   */
+
+String call(String next_version='1000', String repo_type='stable') {
+    BigDecimal _next_version
+    if (next_version == null) {
+        _next_version = 1000
+    } else {
+        try {
+            // Test if it's already a number
+            _next_version = new BigDecimal(next_version)
+        } catch (NumberFormatException e) {
+            // Must be a branch name, convert to a number
+            switch (next_version) {
+                case 'master':
+                    _next_version = 1000
+                    break
+                 case 'release/2.2':
+                    _next_version = 2.3
+                    break
+                 case 'release/2.0':
+                    _next_version = 2.1
+                    break
+                 default:
+                    error("Don't know what the latest version is for ${branch}")
+            }
+        }
+    }
+
+    String v = null
+    try {
+        v = sh(label: 'Get RPM packages version',
+               script: '$(command -v dnf) repoquery --repofrompath=daos,' + env.ARTIFACTORY_URL +
+                       '/artifactory/daos-stack-daos-el-8-x86_64-stable-local/' +
+                     ''' --repoid daos --qf %{version}-%{release} --whatprovides 'daos-tests(x86-64) < ''' +
+                                  _next_version + '''' |
+                              rpmdev-sort | tail -1''',
+               returnStdout: true).trim()
+    /* groovylint-disable-next-line CatchException */
+    } catch (Exception e) {
```

Comment:
Since groovylint cares, do we know what exception(s) could be thrown by a shell step failure?
This really is something that should be in the declarative pipeline documentation.

Reply:
I have not come across any documentation on which exceptions `sh` can throw.